### ISSUE TITLE
XMPP Stanza Error 'remote-server-timeout' yield no type and code

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4311]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#4311|
|Type|enhancement|
|Created|2010-02-24 08:41:46Z|
|Branch|https://github.com/twisted/twisted/tree/xmpp-error-4311|

Due to a typo in `STANZA_CONDITIONS`, if a `StanzaError` is created with the condition `remote-server-timeout`, the error type and code attributes are not set.

<details><summary>Searchable metadata</summary>

```
trac-id__4311 4311
type__enhancement enhancement
reporter__ralphm ralphm
priority__normal normal
milestone__ 
branch__branches_xmpp_error_4311 branches/xmpp-error-4311
branch_author__ralphm ralphm
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1267000906000000 1267000906000000
changetime__1267083964000000 1267083964000000
version__None None
owner__ralphm ralphm
cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm commented|
|-|-|

(In [28515]) Branching to 'xmpp-error-4311'

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm commented|
|-|-|

(In [28517]) Fix typo in STANZA_CONDITIONS, fix spacing and add some docstrings.

Addresses [#4311](https://github.com/twisted/twisted/issues/4311).

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)|@ralphm removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess removed owner|
|-|-|

Thanks for the patch, ralphm. A few comments:

 * In ```test_typeRemoteServerTimeout```, use assertEquals (and in general the plural form of asserts)
 * If you're going to revamp ```test_appConditionMultiple``` use the plural asserts there as well.

Other than that, looks good to merge!

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @ralphm|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_7"></a>|@ralphm commented|
|-|-|

(In [28529]) Address review comments re [#4311](https://github.com/twisted/twisted/issues/4311).

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_8"></a>|@ralphm set status to `closed`|
|-|-|

(In [28530]) Merge xmpp-error-4311: Fix typo preventing remote-server-timeout to get a stanza error type.

Author: ralphm
Reviewer: jesstess
Fixes: [#4311](https://github.com/twisted/twisted/issues/4311)
